### ISSUE TITLE
Pause button for multiplayer 'admin'

### ISSUE THREAD

**Issue by jlederman** [OpenRCT2/OpenRCT2#14503]
Please add the ability for clients with the 'admin' permission to pause the game in multiplayer. 

Expectation: "Toggle Pause" is listed as an option in the in-game Groups editor. Any client with the 'admin' permission can pause the game in multiplayer. 

Context: I'm running a server in headless mode on Debian. Both myself as a client and the server users are set as 'admin' in game, but there's no pause or time control buttons showing up in my client. The only way I can pause is with `context.executeAction("pausetoggle", true)` in the server console.

The idea is the pause button (and other game speed options) will display in the HUD and be usable depending on the permission.

Thank you.

**Comment by janisozaur** [comment]
the pause button as a client is a bit more problematic than a simple update, I believe - we process actions in ticks, but if the game is paused, the new tick won't ever come.

**Comment by janisozaur** [comment]
Fix for this should also consider https://github.com/OpenRCT2/OpenRCT2/issues/11638

**Comment by duncanspumpkin** [comment]
You will have to redesign the whole network architecture of the game to have a secondary tick. This is by no means easy to do.

**Comment by IntelOrca** [comment]
Does build while paused work in multiplayer?

**Comment by Broxzier** [comment]
Yes, building while paused seems to work fine while the game is paused in multiplayer.

![](https://i.imgur.com/Cd9Y5BU.jpg)

There is a 'Toggle Pause' permission, which would be more fitting for this. Note the pause button only appearing for the server, and it is functional, but not bound to this permission.

**Comment by janisozaur** [comment]
does it work when you try executing actions from client side?

**Comment by Broxzier** [comment]
@janisozaur Yes, that all seems to work fine.

**Comment by kscheel** [comment]
The last time I tested this, building while the game did not really work 🤔 yeah, this is still the case for me, just tested this with the current dev build:

When I connect to my headless server, enable the cheat to allow building in pause mode and try to build sth., it looks like nothing has happened. But as soon as I unpause the game I can see the changes I made. I think building while the game is paused only really works for the host (if somebody is playing as host) and not for clients.

@jlederman Until this feature officially implemented, maybe my Plug-In is useful to you: [OpenRCT2-PauseToggle-For-Multiplayer](https://github.com/kscheel/OpenRCT2-PauseToggle-For-Multiplayer)

**Comment by ZehMatt** [comment]
> The last time I tested this, building while the game is paused did not really work 🤔 yeah, this is still the case for me, just tested this with the current dev build:
> 
> When I connect to my headless server, enable the cheat to allow building in pause mode and try to build sth., it looks like nothing has happened. But as soon as I unpause the game I can see the changes I made. I think building while the game is paused only really works for the host (if somebody is playing as host) and not for clients.
> 
> I wonder why this seems to work for @Broxzier though 🤔 In his screenshot the changes seem to show up for the client 🤷‍♂️
> 
> @jlederman Until this feature is officially implemented, my Plug-In might be useful to you: [OpenRCT2-PauseToggle-For-Multiplayer](https://github.com/kscheel/OpenRCT2-PauseToggle-For-Multiplayer)

I'm able to reproduce this issue, I'll have a look into it.
